Source/Core/Core/HW/DVD/DVDThread.cpp

```diff
@@ -358,6 +363,37 @@ static void FinishRead(u64 id, s64 cycles_late)
                                        buffer);
 }
 
+static void PatchReadRequest(std::vector<u8> &buffer, u64 offset) {
+  if(!s_disc_patches)
+    return;
+
+  u64 end_offset = offset + buffer.size();
+
+  // Scan through patches which might overlap our current read request and apply them
+  for (auto it = s_disc_patches->begin(); it != s_disc_patches->end(); it++) {
+    if (it->first > end_offset) // Patch starts after the end of the read request
+    {
+      return;
+    }
+    else if (it->first >= offset) // Patch starts inside the read request
+    {
+      u64 start = it->first - offset;
+      u32 old_data;
+      u32 new_data;
+      std::memcpy(&old_data, buffer.data() + start, 4);
+      std::memmove(buffer.data() + start, it->second.data(), std::min(it->second.size(), buffer.size() - start));
+      std::memcpy(&new_data, buffer.data() + start, 4);
+      INFO_LOG(DVDINTERFACE, "patch applied at %08lx - old: %08x new: %08x", offset + start, old_data, new_data);
```

Comment:
Don't we need to byteswap here in order for the values to make sense?

Reply:
I just removed the old/new all-together. it was wrong in various ways.